pkg/cliplugins/bind/plugin/bind.go

```diff
@@ -0,0 +1,176 @@
+	// to which the user wants to bind.
+	workspacePath logicalcluster.Name
+	// Name of the APIExport.
+	apiexportName string
+	// APIExportRef is the argument accepted by the command. It contains the
+	// reference to where APIExport exists. For ex: <absolute_ref_to_workspace>:<apiexport>.
+	APIExportRef string
+	// Name of the APIBinding.
+	APIBindingName string
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+
+	cmd.Flags().StringVar(&b.APIBindingName, "name", b.APIBindingName, "Name of the APIBinding to create.")
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.APIExportRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.APIExportRef == "" {
+		return errors.New("`root:ws:apiexport_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.APIExportRef, "root") || !logicalcluster.New(b.APIExportRef).IsValid() {
+		return fmt.Errorf("fully qualified reference to workspace where APIExport exists is required. The format is `root:<ws>:<apiexport>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	b.workspacePath, b.apiexportName = logicalcluster.New(b.APIExportRef).Split()
+
+	// if apibindingName is not provided, default it to <apiExportname>.
+	apiBindingName := b.APIBindingName
+	if apiBindingName == "" {
+		apiBindingName = b.apiexportName
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
+	}
+
+	binding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: apiBindingName,
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       b.workspacePath.String(),
+					ExportName: b.apiexportName,
+				},
+			},
+		},
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	createdBinding, err := kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "apibinding %s created. Waiting to successfully bind ...\n", binding.Name); err != nil {
+		return err
+	}
+
+	// wait for phase to be bound
```

Comment:
You could make this a flag `--wait` that defaults to true

Reply:
Just curious, what would be the use case of having `--wait=false`. The whole bind command's purpose seems to be to make sure that the binding is successfully created. That's why, made this to be a part of the command itself.

Reply:
👍 for being part of the command. I mostly suggested it because there are other commands in kubectl that have a `--wait` flag.